docs(astro): Add multi-framework example for $signInStore and $signUpStore (#1435)
clerk/clerk-docs: docs/references/astro/sign-in-store.mdx

# `$signInStore`

> The `$signInStore` store provides a convenient way to access the [`SignIn`](https://clerk.com/docs/references/javascript/sign-in/sign-in) object, which allows you to check the current state of a sign-in. This is also useful for creating a custom sign-in flow.

Reviewer: ```suggestion
The `$signInStore` store provides a convenient way to access the [`SignIn`](/docs/references/javascript/sign-in/sign-in){{ target: '_blank' }} object, which allows you to check the current state of a sign-in. This is also useful for creating a custom sign-in flow.
```

## How to use the `$signInStore` store

### Check the current state of a sign-in

The following example demonstrates how to use the `$signInStore` store to check the current state of a sign-in.

<CodeBlockTabs options={['React', 'Vue', 'Svelte']}>

>   ```tsx {{ prettier: false, filename: 'sign-in-step.tsx' }}

Reviewer: Is there a reason we want to turn off prettier for the React code block?

```suggestion
  ```tsx {{ filename: 'sign-in-step.tsx' }}
```

  }
  ```

  ```vue {{ prettier: false, filename: 'sign-in-step.vue' }}
  <script setup>
  import { useStore } from '@nanostores/vue';
  import { $signInStore } from '@clerk/astro/client';

  const signIn = useStore($signInStore);
  </script>

  <template>
    <div v-if="signIn === undefined">
      <!-- Add logic to handle loading state -->
    </div>
    <div v-else>
      <div>The current sign in attempt status is {{ signIn.status }}.</div>
    </div>
  </template>
  ```

  ```svelte {{ prettier: false, filename: 'sign-in-step.svelte' }}
  <script>
  // The $ prefix is reserved in Svelte for its own reactivity system.
  // Alias the imports to avoid conflicts.
  import { $signInStore as signIn } from '@clerk/astro/client';
  </script>

  {#if $signIn === undefined}
      <!-- Add logic to handle loading state -->
  {:else}
      <div>The current sign in attempt status is {$signIn.status}.</div>
  {/if}
  ```
</CodeBlockTabs>

The possible values for the `status` property of the `SignIn` resource are listed [here](/docs/references/javascript/sign-in/sign-in#properties).

### Create a custom sign-in flow

> The `$signInStore` store can also be used to build fully custom sign-in flows, if Clerk's pre-built components don't meet your specific needs or if you require more control over the authentication flow. Different sign-in flows include email and password, email and phone codes, email links, and multifactor (MFA). To learn more about using the `$signInStore` store to create custom flows, check out the [custom flow guides](/docs/custom-flows/overview)

Reviewer: ```suggestion
The `$signInStore` store can also be used to build fully custom sign-in flows, if Clerk's pre-built components don't meet your specific needs or if you require more control over the authentication flow. Different sign-in flows include email and password, email and phone codes, email links, and multifactor (MFA). To learn more about using the `$signInStore` store to create custom flows, check out the [custom flow guides.](/docs/custom-flows/overview)
```
